### ISSUE TITLE
Expend beyond braces, add smart alternatives to other motions

### ISSUE THREAD

**Issue by alerque** [Konfekt/vim-smartbraces#5]
Not that the decision to use or not use this plugin (see #4) has been resolved for me and the mental process is not "do I want to go to the beginning of the _next_ object like this (`{`) or the end of _this_ one (`g{`), I've started to want the same logic to apply to other motions.

Notably sentence wise motion (`(`) suffers from the same issue. The motion makes sense for deleting a sentence `d)`, but there are other times when I don't want the start of the next sentence, I want to move to the last character of the current one.

I don't know if any of this code is reusable for that purpose, but if it is adding variants of the parenthesis motions that correspond to the smart brace motions would be a nice addition.

**Comment by Konfekt** [comment]
Yes, thanks for the suggestion. Likely this plug-in will then deserve another name though.) 

I borrowed the mapping suggested at https://github.com/Konfekt/vim-sentence-chopper#hints to make `)` jump to the end of the sentence. 

If you want `d)` to work as before, add
```vim
onoremap <expr> )       '<esc>' . '"' . v:register . v:operator . v:count1 . (v:operator ==# 'd' ? ')' : ':<c-u>call search('\v\C%(%([^[:digit:]IVX]\|[)''"])\zs[.]\|[!?])[[:space:])''"]\|[.!?]$','W')<CR>')
```
to your `vimrc`.
